### PR TITLE
Update Android SDKs, Java JDK and action versions to newer versions

### PR DESCRIPTION
**Pull request by jonasanker**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -167,7 +167,7 @@ jobs:
         uses: actions/checkout@v2.4.0
 
       - name: Setup Flutter environment
-        uses: subosito/flutter-action@v1.5.3
```
Comment (fremartini) [inline .github/workflows/build.yml:170]: i think we should pin the version to avoid stuff breaking because of new versions.

**Review by fremartini — CHANGES_REQUESTED** [review]
i think we should pin the version to avoid stuff breaking because of new versions.

File: .github/workflows/build.yml (hunk 1 above)
Comment (jonasanker) [reply .github/workflows/build.yml:170]: I don't disagree as such. However we also be fine to lock into just the major or minor version as you wouldn't normally expect breaking behaviour between patches .

**Comment by jonasanker** [comment]
> i think we should pin the version to avoid stuff breaking because of new versions.

I don't disagree as such. However we also be fine to lock into just the major or minor version as you wouldn't normally expect breaking behaviour between patches

**Comment by fremartini** [comment]
> > i think we should pin the version to avoid stuff breaking because of new versions.
> 
> I don't disagree as such. However we also be fine to lock into just the major or minor version as you wouldn't normally expect breaking behaviour between patches.
> 
> Could we maybe just lock in in the minor version?

Minor versions would be fine. Expectations and reality have a tendency to not align

**Comment by jonasanker** [comment]
> > > i think we should pin the version to avoid stuff breaking because of new versions.
> > 
> > 
> > I don't disagree as such. However we also be fine to lock into just the major or minor version as you wouldn't normally expect breaking behaviour between patches.
> > Could we maybe just lock in in the minor version?
> 
> Minor versions would be fine. Expectations and reality have a tendency to not align

It seems you can not refer to a minor version. Only major versions or absolute versions are supported, thus I have changed all actions to refer their latest version.

@fremartini Please re-review

**Review by fremartini — APPROVED** [review]

**Pull request merged**